torch/distributed/_state_dict_utils.py

```diff
@@ -423,10 +429,23 @@ def unpin_memory(t):
         else:
             return torch.empty(*tuple(obj.size()), dtype=obj.dtype)
 
+    def dtensor_func(
+        obj: DTensor,
+        pg: Optional[dist.ProcessGroup],
+        device: Optional[torch.device],
+        _: Any,
+    ) -> DTensor:
+        if len(obj.size()) == 0:
+            return obj
+
+        ret = cast(DTensor, copy.deepcopy(obj.to(device="cpu")))
```

Comment:
Does this deepcopy the local_tensor as well? It may be safer to manually instantiate the DTensor

Just want to make sure we're not doing duplicate allocations/copies

Reply:
We will replace `_local_tensor` by using `tensor_func()` in the next two lines. So `_local_tensor` will not be the same.

Reply:
Change it to deepcopy cpu DTensor to avoid duplicate copies.